### PR TITLE
fix string escaping

### PR DESCRIPTION
**Pull request by vobloeb**
Sequences like `"\."` and `"\@"` are wacky within single quoted strings, there must be either `"."` or `"\\."`

### CODE REVIEW

File: pkgs/development/libraries/libcutl/default.nix [hunk 1]
```diff
@@ -18,7 +18,7 @@ stdenv.mkDerivation rec {
     license = licenses.mit;
   };
 
-  majmin = builtins.head ( builtins.match "([[:digit:]]\.[[:digit:]]+)\.*" "${version}" );
+  majmin = builtins.head ( builtins.match "([[:digit:]]\\.[[:digit:]]+)\\.*" "${version}" );
```
Comment (lourkeur) [inline pkgs/development/libraries/libcutl/default.nix:21]:
```suggestion
  majmin = builtins.head ( builtins.match "([[:digit:]]\\.[[:digit:]]+).*" "${version}" );
```

That should fix the evaluation. It does go to show that these constructs are dodgy. "\\.*" would be too strict to match the rest of the sequence, in this instance ".0"

**Review by taku0 — APPROVED** [review]

**Comment by taku0** [comment]
Please squash `Update pkgs/development/libraries/libcutl/default.nix` and `libcutl: fix string escaping`, then I will merge this.

**Comment by taku0** [comment]
@kvtb ping

**Comment by kvtb** [comment]
I cannot, I have no full nixpkgs git repo locally (checking out will eat monthly packet of mobile Internet) and GitHub web ui has no squash

**Comment by taku0** [comment]
@kvtb could you [allow us to edit your branch](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)? I will try to squash and (force-)push if I can.

**Comment by kvtb** [comment]
> @kvtb could you [allow us to edit your branch](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)? I will try to squash and (force-)push if I can.

I see this checkbox on my other two PRs, but not on this one.
And `kvtb:patch-2` above is just text, not link to https://github.com/kvtb/nixpkgs/tree/patch-2. In another PR, `kvtb:patch-3` is a link.
May be better kill this PR and make new one.

**Comment by kvtb** [comment]
Interesting, https://github.com/kvtb/nixpkgs/tree/patch-2 does not have commits from this PR.
So this PR is effectively branchless.

**Comment by lourkeur** [comment]
true.

I grabbed refs/pull/108086/head an went to rebase. Thank you for your work!

**Pull request closed without merging**